### PR TITLE
feat: engineering health dashboard

### PR DESCRIPTION
**Pull request by datphamcode295**
#### What's this PR do?

- [x] Engineering health for dashboard

#### What are the relevant Git tickets?
https://www.notion.so/dwarves/API-fetch-Engineering-health-score-report-623b31de13e5461a8ef3bd69667c9c54

#### Screenshots (if appropriate)
image

#### Request
```
curl --location --request GET 'http://localhost:8200/api/v1/dashboards/engineering-healths?projectID=8dc3be2e-19a4-4942-8a79-56db391a0b15' \
--header 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2OTkzMjExNDIsImlkIjoiMjY1NTgzMmUtZjAwOS00YjczLWE1MzUtNjRjM2EyMmU1NThmIiwiYXZhdGFyIjoiaHR0cHM6Ly9zMy1hcC1zb3V0aGVhc3QtMS5hbWF6b25hd3MuY29tL2ZvcnRyZXNzLWltYWdlcy81MTUzNTc0Njk1NjYzOTU1OTQ0LnBuZyIsImVtYWlsIjoidGhhbmhAZC5mb3VuZGF0aW9uIiwicGVybWlzc2lvbnMiOlsiZW1wbG95ZWVzLnJlYWQiXSwidXNlcl9pbmZvIjpudWxsfQ.GENGPEucSUrILN6tHDKxLMtj0M0REVMUPC7-XhDMpGM' \
--data-raw ''
```

#### Response
```
{
    "data": {
        "average": [
            {
                "quarter": "Q1/2022",
                "avg": 3.9,
                "trend": 0
            },
            {
                "quarter": "Q2/2022",
                "avg": 3.8,
                "trend": -2.56
            }
        ],
        "groups": {
            "delivery": [
                {
                    "quarter": "Q1/2022",
                    "avg": 4,
                    "trend": 0
                },
                {
                    "quarter": "Q2/2022",
                    "avg": 3,
                    "trend": -25
                }
            ],
            "quality": [
                {
                    "quarter": "Q1/2022",
                    "avg": 2,
                    "trend": 0
                },
                {
                    "quarter": "Q2/2022",
                    "avg": 4,
                    "trend": 100
                }
            ],
            "collaboration": [
                {
                    "quarter": "Q1/2022",
                    "avg": 3,
                    "trend": 0
                },
                {
                    "quarter": "Q2/2022",
                    "avg": 3,
                    "trend": 0
                }
            ],
            "feedback": [
                {
                    "quarter": "Q1/2022",
                    "avg": 4,
                    "trend": 0
                },
                {
                    "quarter": "Q2/2022",
                    "avg": 4,
                    "trend": 0
                }
            ]
        }
    }
}
```

### CODE REVIEW

File: pkg/view/dashboard.go [hunk 1]
```diff
@@ -151,3 +152,123 @@ func calculateActionItemReportTrend(previous *model.ActionItemReport, current *m
...
+			count++
+			quarter = g.Quarter
+
+			if count > 4 {
+				break
+			}
+		}
+
+		switch g.Area {
+		case "Delivery performance":
```
Comment (thanhpn) [inline pkg/view/dashboard.go:225]: Move these string to const

File: pkg/view/dashboard.go [hunk 2]
```diff
@@ -151,3 +152,123 @@ func calculateActionItemReportTrend(previous *model.ActionItemReport, current *m
...
+			if len(rs.Delivery) < 4 {
```
Comment (thanhpn) [inline pkg/view/dashboard.go:226]: Why we check length smaller than 4 ?

File: pkg/view/dashboard.go (hunk 1 above)
Comment (datphamcode295) [reply pkg/view/dashboard.go:225]: updated, thank you

File: pkg/view/dashboard.go (hunk 2 above)
Comment (datphamcode295) [reply pkg/view/dashboard.go:226]: To make sure have less than 4 quarter, but I think the condition
```
if count > 4 {
				break
			}
```
is enough, thank you

**Review by thanhpn — APPROVED** [review]

**Pull request merged**